bioio/writers/ome_zarr_writer.py

```diff
@@ -1,431 +1,478 @@
+    return out
+
+
+def default_axes(
+    names: List[str], types: List[str], units: List[Optional[str]]
+) -> List[dict]:
+    """
+    Build axes metadata list from provided names, types, and units.
+
+    Parameters
+    ----------
+    names : List[str]
+        Axis names.
+    types : List[str]
+        Axis types.
+    units : List[Optional[str]]
+        Axis units (or None).
+
+    Returns
+    -------
+    List[dict]
+        List of axis metadata dictionaries.
+    """
+
+    # TODO: should this be a constant or a util or something?
+    axes = []
+    for n, t, u in zip(names, types, units):
+        entry = {"name": n, "type": t}
+        if u:
+            entry["unit"] = u
+        axes.append(entry)
+    return axes
+
+
+class OMEZarrWriter:
+    """
+    OMEZarrWriter is a fully compliant OME-Zarr v0.5.0 writer built on Zarr v3 stores.
+    """
+
+    def __init__(
+        self,
+        store: Union[str, zarr.storage.StoreLike],
+        shape: Tuple[int, int, int, int, int],
+        dtype: Union[np.dtype, str],
+        axes_names: List[str] = ["t", "c", "z", "y", "x"],
+        axes_types: List[str] = ["time", "channel", "space", "space", "space"],
+        axes_units: List[Optional[str]] = [None, None, None, None, None],
+        axes_scale: List[float] = [1.0, 1.0, 1.0, 1.0, 1.0],
+        scale_factors: Tuple[int, int, int, int, int] = (1, 1, 2, 2, 2),
+        num_levels: Optional[int] = None,
+        chunks: Union[str, Tuple[int, ...], List[Tuple[int, ...]]] = "auto",
+        shards: Optional[Union[Tuple[int, ...], List[Tuple[int, ...]]]] = None,
+        compressor: Optional[BloscCodec] = None,
+        image_name: str = "Image",
+        channel_names: Optional[List[str]] = None,
+        channel_colors: Optional[List[str]] = None,
+        creator_info: Optional[dict] = None,
+    ):
         """
-        Constructor.
+        Initialize writer and automatically build axes and channel metadata.
 
         Parameters
         ----------
-        uri: biob.types.PathLike
-            The URI or local path for where to save the data.
+        store : Union[str, zarr.storage.StoreLike]
+            Path or Zarr Store for output.
+        shape : Tuple[int, int, int, int, int]
+            Base image shape (T, C, Z, Y, X).
+        dtype : Union[np.dtype, str]
+            NumPy dtype of the image data.
+        axes_names : List[str]
+            Axis names.
+        axes_types : List[str]
+            Axis types.
+        axes_units : List[Optional[str]]
+            Axis units.
+        axes_scale : List[float]
+            Physical scale per axis.
+        scale_factors : Tuple[int, int, int, int, int]
+            Downsampling factors per axis.
+        num_levels : Optional[int]
+            Maximum number of pyramid levels.
+        chunks : Union[str, Tuple[int, ...], List[Tuple[int, ...]]]
+            Chunk specification or "auto".
+        shards : Optional[Union[Tuple[int, ...], List[Tuple[int, ...]]]]
```

Comment:
Can we put the defaults in here? Still kinda feeling like this writer could have a factory method for users that don't have the brain bandwidth or care enough to optimize for the best zarr share for their usage. The simplicity level of at least the .ome.tiff writer is what I'm thinking for the factory method.

Reply:
Agreed. I think that's another PR though, this ones getting pretty large.